### DISCUSSION TITLE
Kibana stops working after a few weeks of running

### DISCUSSION THREAD

**Discussion by SackOfHacks** [Security-Onion-Solutions/securityonion#3093] [Unsupported Versions]
I have had 3 installs that Kibana quit working on after about 3 weeks, showing the error code:

![image](https://user-images.githubusercontent.com/16657192/108899726-fdb65180-75d5-11eb-9ef0-cba019c0a381.png)

**Comment by Acewiza** [comment]
I like the one that says "unknown error in the system."  Two weeks seems to be about the max for me.  Not always the same failure, but almost like clockwork at around the 2-week mark, some service errors out and refuses to restart.  It was steno a couple days ago.

**Comment by TOoSmOotH** [comment]
And the logs say?

**Reply by Acewiza** [reply]
It's not a big deal, i.e. time sensitive.  Reboot takes a few seconds every 2 weeks.  Not sure where my annoyance level for looking at logs starts in this case.  Probably in the less-than-a-week area when I'm less busy with other things.  ;-)

**Reply by SackOfHacks** [reply]
When I run the so-status, all comes back green. The security Onion landing page works, but anything related to Elasticsearch (Kibana, Alerts, etc.) fails with an error 500. A reboot of the system does not help.

What logs can I provide to help with this?

**Reply by TOoSmOotH** [reply]
/opt/so/log has all the logs in it. You would want to check the Kibana and the Elasticsearch logs to see if they are throwing any errors.

**Reply by deannick** [reply]
Im having what to be the same issues.  Here is the Kibana log.


{"type":"log","@timestamp":"2021-02-24T12:56:07Z","tags":["warning","plugins","kibanaUsageCollection","application-usage"],"pid":292,"message":"Failed to rollup transactional to daily entries"}
{"type":"log","@timestamp":"2021-02-24T12:56:07Z","tags":["warning","plugins","kibanaUsageCollection","application-usage"],"pid":292,"message":"{ Error: Request Timeout after 90000ms\n    at /usr/share/kibana/node_modules/elasticsearch/src/lib/transport.js:397:9\n    at Timeout.<anonymous> (/usr/share/kibana/node_modules/elasticsearch/src/lib/transport.js:429:7)\n    at ontimeout (timers.js:436:11)\n    at tryOnTimeout (timers.js:300:5)\n    at listOnTimeout (timers.js:263:5)\n    at Timer.processTimers (timers.js:223:10)\n  status: undefined,\n  displayName: 'RequestTimeout',\n  message: 'Request Timeout after 90000ms',\n  body: undefined,\n  isBoom: true,\n  isServer: true,\n  data: null,\n  output:\n   { statusCode: 503,\n     payload:\n      { statusCode: 503,\n        error: 'Service Unavailable',\n        message: 'Request Timeout after 90000ms' },\n     headers: {} },\n  reformat: [Function],\n  [Symbol(SavedObjectsClientErrorCode)]: 'SavedObjectsClient/esUnavailable' }"}
{"type":"log","@timestamp":"2021-02-24T13:24:38Z","tags":["info","plugins-service"],"pid":292,"message":"Plugin \"telemetryManagementSection\" has been disabled since the following direct or transitive dependencies are missing or disabled: [telemetry]"}
{"type":"log","@timestamp":"2021-02-24T13:24:38Z","tags":["info","plugins-service"],"pid":292,"message":"Plugin \"telemetry\" is disabled."}
{"type":"log","@timestamp":"2021-02-24T13:24:38Z","tags":["info","plugins-service"],"pid":292,"message":"Plugin \"visTypeXy\" is disabled."}
{"type":"log","@timestamp":"2021-02-24T13:24:39Z","tags":["warning","config","deprecation"],"pid":292,"message":"You should set server.basePath along with server.rewriteBasePath. Starting in 7.0, Kibana will expect that all requests start with server.basePath rather than expecting you to rewrite the requests in your reverse proxy. Set server.rewriteBasePath to false to preserve the current behavior and silence this warning."}
{"type":"log","@timestamp":"2021-02-24T13:24:40Z","tags":["info","plugins-system"],"pid":292,"message":"Setting up [37] plugins: [usageCollection,telemetryCollectionManager,kibanaUsageCollection,newsfeed,mapsLegacy,kibanaLegacy,timelion,share,legacyExport,esUiShared,charts,bfetch,expressions,data,home,console,apmOss,management,indexPatternManagement,advancedSettings,savedObjects,visualizations,visualize,visTypeVislib,visTypeVega,visTypeTimeseries,visTypeTimelion,visTypeTagcloud,visTypeTable,visTypeMetric,visTypeMarkdown,tileMap,regionMap,inputControlVis,discover,dashboard,savedObjectsManagement]"}
{"type":"log","@timestamp":"2021-02-24T13:24:40Z","tags":["info","savedobjects-service"],"pid":292,"message":"Waiting until all Elasticsearch nodes are compatible with Kibana before starting saved objects migrations..."}
{"type":"log","@timestamp":"2021-02-24T13:24:40Z","tags":["info","savedobjects-service"],"pid":292,"message":"Starting saved objects migrations"}
{"type":"log","@timestamp":"2021-02-24T13:24:40Z","tags":["warning","savedobjects-service"],"pid":292,"message":"Unable to connect to Elasticsearch. Error: [circuit_breaking_exception] [parent] Data too large, data for [<http_request>] would be [622064624/593.2mb], which is larger than the limit of [597688320/570mb], real usage: [622064624/593.2mb], new bytes reserved: [0/0b], usages [request=0/0b, fielddata=0/0b, in_flight_requests=0/0b, accounting=78879336/75.2mb], with { bytes_wanted=622064624 & bytes_limit=597688320 & durability=\"PERMANENT\" }"}

**Reply by cm-ops** [reply]
`Unable to connect to Elasticsearch. Error: [circuit_breaking_exception] [parent] Data too large, data for [<http_request>] would be [622064624/593.2mb], which is larger than the limit of [597688320/570mb]`

Check your esheap setting in your manager minion pillar file at /opt/so/saltstack/local/pillar/minions/$minion.sls? Try and increase the setting and restart Elasticsearch and see if that helps.

**Reply by SackOfHacks** [reply]
I see tons of these alerts:
![image](https://user-images.githubusercontent.com/16657192/110856324-ba660f00-8274-11eb-889e-0bd3e5016835.png)


Tried increasing the heap size to 8g (on a 16g system) without success. Full disclosure, I replaced the ES_HEAP_SIZE variable with '8g' in my .sls file, might not be the right way.

Anything else I can try?
